### ISSUE TITLE
Leader election without a timer

### ISSUE THREAD

**Issue by hdhillan** [faust-streaming/faust#321]
Based on the documentation, it seems that Faust only supports leader election for periodic agents. I understand that the `concurrency` keyword argument to `app.agent` can be used to specify the number of agent instances to start *per worker*. However, I would like to guarantee that I have *exactly one* agent (globally, *across all workers*) consume certain topics. My goal is that if one worker fails, an instance of the agent running another worker will take over consuming the topic

Is this currently possible with Faust? Should I be using a different tool to accomplish this?

**Comment by wbarnha** [comment]
> My goal is that if one worker fails, an instance of the agent running another worker will take over consuming the topic

I don't think we currently have this functionality. My advice is to run your Consumer in a separate application if you're concerned about it regularly crashing.
